lib/icalendar/parser.rb

```diff
@@ -23,6 +23,7 @@ def parse
       read_in_data
       components = []
       while (fields = next_fields)
+        component = component_class.new
```

Comment:
Instead of creating a new `component` each time through this loop, should we switch line 28 to be `components << parse_component(component_class.new)`?